doc/source/whatsnew/v0.18.1.txt

```diff
@@ -76,8 +76,7 @@ API changes
 
 
 
-
-
+- In ``read_csv``, the Python parser now raises an ``EmptyDataError`` in response to empty columns or header (:issue:`12506`)
```

Comment:
The previous user experience used to be `StopIteration`, are you saying that we continue to raise that?